ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -144,12 +144,21 @@ private void onBlockAdded(final BlockAddedEvent blockAddedEvent, final Blockchai
     }
   }
 
-  private void handleNewBlockFromNetwork(final EthMessage message) {
+  void broadcastBlock(final Block block, final UInt256 difficulty) {
```

Comment:
Is there a reason this is package private instead of private?

I'd also suggest as you move forward to extract this out into a separate `BlockBroadcaster` class. It just provides a better separation of concerns and will let us write a more focussed unit test for the approach to broadcasting blocks.  Happy enough for it to live here for now though.